tests/functional/z_cluster/test_performance_profile_validation.py

```diff
@@ -36,28 +38,71 @@
 @pytest.mark.polarion_id("OCS-5656")
 @pytest.mark.polarion_id("OCS-5657")
 class TestProfileDefaultValuesCheck(ManageTest):
-    def test_validate_cluster_resource_profile(self):
+    @pytest.mark.parametrize(
+        argnames=["perf_profile"],
+        argvalues=[
+            pytest.param(*["performance"]),
+            pytest.param(*["lean"]),
+            pytest.param(*["balanced"]),
+        ],
+    )
+    @ignore_leftovers
+    def test_validate_cluster_resource_profile(self, perf_profile):
         """
         Testcase to validate osd, mgr, mon, mds and rgw pod memory and cpu values
         are matching with the predefined set of values post profile updation
 
         """
         pv_pod_obj = []
+        namespace = config.ENV_DATA["cluster_namespace"]
         log.info("Obtaining the performance profile values from the cluster")
         storage_cluster_name = config.ENV_DATA["storage_cluster_name"]
         storage_cluster = StorageCluster(
             resource_name=storage_cluster_name,
             namespace=config.ENV_DATA["cluster_namespace"],
         )
-        performance_profile = storage_cluster.data["spec"]["resourceProfile"]
-        if performance_profile == constants.PERFORMANCE_PROFILE_LEAN:
+        self.perf_profile = perf_profile
+
+        try:
+            exist_performance_profile = storage_cluster.data["spec"]["resourceProfile"]
+        except KeyError:
+            exist_performance_profile = None
+            pass
+        if exist_performance_profile == self.perf_profile:
+            log.info("Performance profile is same as profile that is already present")
+        else:
+            ptch = f'{{"spec": {{"resourceProfile":"{self.perf_profile}"}}}}'
+            ptch_cmd = (
+                f"oc patch storagecluster {storage_cluster.data.get('metadata').get('name')} "
+                f"-n {namespace}  --type merge --patch '{ptch}'"
+            )
+            run_cmd(ptch_cmd)
+            log.info("Verify storage cluster is on Ready state")
+            verify_storage_cluster()
+
+            # Sleep for 120 seconds for the profile changes to reflect
+            time.sleep(600)
```

Comment:
Here I would use `TimeoutSampler` to wait for the profile to change to `reflect`.

Reply:
Added Timeout sampler with 5 mins sleep time, this is required though cluster shows ready state it takes some times for the pod to actuall reflect those values